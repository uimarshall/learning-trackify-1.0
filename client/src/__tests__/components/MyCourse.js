import '@testing-library/jest-dom';

import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import MyCourse from '../../components/MyCourse';

jest.mock('../../components/MyCourse', () => () => <div id="MyCourse">MyCourse</div>);
describe('MyCourse Component', () => {
  test('renders MyCourse in the DOM', () => {
    const { container } = render(<MyCourse />);
    const mockComponent = container.querySelector('div#MyCourse');

    expect(mockComponent).toBeInTheDocument();
    expect(mockComponent).not.toHaveTextContent(/^arbitrary$/); // to match the whole content
    expect(mockComponent).not.toHaveTextContent(/fire$/i); // to use case-insensitive match
    expect(mockComponent).toHaveTextContent('MyCourse');
  });
});

describe('MyCourse Component', () => {
  afterEach(cleanup);

  it('should take a snapshot for MyCourse Component', () => {
    const asFragment = renderer.create(<MyCourse />).toJSON()
    expect(asFragment).toMatchSnapshot();
  });
});