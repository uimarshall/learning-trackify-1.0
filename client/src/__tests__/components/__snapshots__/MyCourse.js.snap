// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCourse Component should take a snapshot for MyCourse Component 1`] = `
<div
  id="MyCourse"
>
  MyCourse
</div>
`;
